benchmarks/benchmarks/loading.py

```diff
@@ -0,0 +1,186 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+File loading benchmark tests.
+
+Where applicable benchmarks should be parameterised for two sizes of input data:
+  * minimal: enables detection of regressions in parts of the run-time that do
+             NOT scale with data size.
+  * large: large enough to exclusively detect regressions in parts of the
+           run-time that scale with data size. Size should be _just_ large
+           enough - don't want to bloat benchmark runtime.
+
+"""
+
+from iris import AttributeConstraint, Constraint, load, load_cube
+from iris.cube import Cube
+from iris.fileformats.um import structured_um_loading
+
+from .generate_data import BENCHMARK_DATA, REUSE_DATA, run_function_elsewhere
+from .generate_data.um_files import create_um_files
+
+
+class LoadAndRealise:
+    params = [
+        [(2, 2, 2), (1280, 960, 5)],
+        [False, True],
+        ["FF", "PP", "NetCDF"],
+    ]
+    param_names = ["xyz", "compressed", "file_format"]
+
+    def setup_cache(self) -> dict:
+        file_type_args = self.params[2]
+        file_path_dict = {}
+        for xyz in self.params[0]:
+            file_path_dict[xyz] = {}
+            x, y, z = xyz
+            for compress in self.params[1]:
+                file_path_dict[xyz][compress] = create_um_files(
+                    x, y, z, 1, compress, file_type_args
+                )
+        return file_path_dict
+
+    def setup(
+        self,
+        file_path_dict: dict,
+        xyz: tuple,
+        compress: bool,
+        file_format: str,
+    ) -> None:
+        self.file_path = file_path_dict[xyz][compress][file_format]
+        self.cube = self.load()
+
+    def load(self) -> Cube:
+        return load_cube(self.file_path)
+
+    def time_load(self, _, __, ___, ____) -> None:
+        _ = self.load()
+
+    def time_realise(self, _, __, ___, ____) -> None:
+        # Cache the original data object and re-apply after realisation, which
+        #  restores original state for the next repeat. The cache/apply steps
+        #  add negligible time to the benchmark result.
+        data_original = self.cube.core_data()
+        _ = self.cube.data
+        self.cube.data = data_original
```

Comment:
>I think I would use `self.cube.core_data().compute()`

Is your suggestion in place of `_ = self.cube.data`? It's hard to tell! If so, I supposed I was using the syntax that we normally recommend to the user.

>I'm puzzled by the need for this.

This was actually something you originally discovered and rectified. ASV performs re-runs between `self.setup()` calls, meaning that it can only time realisation once, then the data is already realised at the start subsequent re-runs. So we restore the original state before finishing the benchmark so that it's ready for the next re-run (providing that this action doesn't have a big effect on the benchmark result).

Reply:
**Update**: well **_I_** still think it would be simpler to ditch all the comment + explicitly fetch the lazy content + compute it.
But we can discuss.

Reply:
f96480db26e38cafecb2bcdfbc5b5446c0842b11

Reply:
:+1: for explaining the issues nicely !